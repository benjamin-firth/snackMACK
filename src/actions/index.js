export const addPotentialLocation = potentialLocation => ({
  type: 'ADD_POTENTIAL_LOCATION',
  potentialLocation
});

export const removePotentialLocation = potentialLocation => ({
  type: 'REMOVE_POTENTIAL_LOCATION',
  potentialLocation
})

export const chooseCity = (city, lat, long) => ({
  type: 'CHOOSE_CITY',
  city: {
    city,
    lat,
    long
  }
})
