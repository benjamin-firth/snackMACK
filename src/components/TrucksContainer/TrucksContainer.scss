.right-side-box {
  width: 100%;
}

.leaflet-container {
  width: 100%;
  height: 100vh;
}

.mainpage-container {
  background-color: #231E42;
  display: flex;
}

.sidebar {
  background-color: #231E42;
  border-right: 2px solid #83A1B0;
  padding: 20px;
  width: 40%;

  button {
    background-color: #83A1B0;
    border-radius: 5px;
    color: #302D50;
    font-size: 21px;
    margin-bottom: 25px;
    width: 100%;
  }
  
  .inner-scroll-container {
    overflow: scroll;
    height: 83vh;
  }
}