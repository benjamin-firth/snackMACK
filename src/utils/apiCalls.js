export const fetchFoodTrucks = async () => {
  const response = await fetch('https://cors-anywhere.herokuapp.com/http://data.streetfoodapp.com/1.1/schedule/calgary')
  const foodTrucks = await response.json();
  return foodTrucks;
}