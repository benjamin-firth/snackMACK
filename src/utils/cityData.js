export const cityData = {
  calgary: {lat: 51.0, long: -114.0},
  edmonton: {lat: 53.5, long: -113.5},
  halifax: {lat: 44.6, long: -63.6},
  hamilton: {lat: 43.3, long: -80.0},
  ottawa: {lat: 45.4, long: -75.7},
  toronto: {lat: 43.7, long: -79.4},
  vancouver: {lat: 49.3, long: -123.1},
  victoria: {lat: 37.5, long: -144.8},
}




