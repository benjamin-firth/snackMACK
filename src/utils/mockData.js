export const mockPotentialLocations = [{
  desc: "Our food trailer is designed for efficiently producing quality comfort food! We serve baked potatoes loaded with a variety of toppings! BBQ Pulled Pork Tex Mex Mushroom Swiss Classic English Sheppard's Pie",
  isPotentialLocation: true,
  image: undefined,
  location: {
    display: "Prairie Horizon Pop Up Market",
    latitude: 51.19624517,
    longitude: -113.99491753,
    time: 1578762000
  },
  name: "Baked And Loaded"
}, 
{
  desc: "Our food trailer is designed for efficiently producing quality comfort food! We serve baked potatoes loaded with a variety of toppings! BBQ Pulled Pork Tex Mex Mushroom Swiss Classic English Sheppard's Pie",
  isPotentialLocation: true,
  image: undefined,
  location: {
    display: "Prairie Horizon Pop Up Market",
    latitude: 51.19624517,
    longitude: -113.99491753,
    time: 1578762000
  },
  name: "Test Food Truck 2"
}];

export const mockPotentialLocationsAfterToggle = [{
  desc: "Our food trailer is designed for efficiently producing quality comfort food! We serve baked potatoes loaded with a variety of toppings! BBQ Pulled Pork Tex Mex Mushroom Swiss Classic English Sheppard's Pie",
  isPotentialLocation: false,
  image: undefined,
  location: {
    display: "Prairie Horizon Pop Up Market",
    latitude: 51.19624517,
    longitude: -113.99491753,
    time: 1578762000
  },
  name: "Baked And Loaded"
}, 
{
  desc: "Our food trailer is designed for efficiently producing quality comfort food! We serve baked potatoes loaded with a variety of toppings! BBQ Pulled Pork Tex Mex Mushroom Swiss Classic English Sheppard's Pie",
  isPotentialLocation: true,
  image: undefined,
  location: {
    display: "Prairie Horizon Pop Up Market",
    latitude: 51.19624517,
    longitude: -113.99491753,
    time: 1578762000
  },
  name: "Test Food Truck 2"
}]

export const mockFilteredPotentialLocations = {
  desc: "Our food trailer is designed for efficiently producing quality comfort food! We serve baked potatoes loaded with a variety of toppings! BBQ Pulled Pork Tex Mex Mushroom Swiss Classic English Sheppard's Pie",
  image: undefined,
  location: {
    display: "Prairie Horizon Pop Up Market",
    latitude: 51.19624517,
    longitude: -113.99491753,
    time: 1578762000
  },
  name: "Baked And Loaded"
}

export const setTrucksPotentialLocationActionObj = {
  type: 'SET_TRUCKS',
  trucks: {
    desc: "Our food trailer is designed for efficiently producing quality comfort food! We serve baked potatoes loaded with a variety of toppings! BBQ Pulled Pork Tex Mex Mushroom Swiss Classic English Sheppard's Pie",
    image: undefined,
    location: {
      display: "Prairie Horizon Pop Up Market",
      latitude: 51.19624517,
      longitude: -113.99491753,
      time: 1578762000
    },
    name: "Baked And Loaded"
  }
}

export const mockAddPotentialLocationActionObj = {
  type: 'ADD_POTENTIAL_LOCATION',
  potentialLocation: {
    desc: "Our food trailer is designed for efficiently producing quality comfort food! We serve baked potatoes loaded with a variety of toppings! BBQ Pulled Pork Tex Mex Mushroom Swiss Classic English Sheppard's Pie",
    image: undefined,
    location: {
      display: "Prairie Horizon Pop Up Market",
      latitude: 51.19624517,
      longitude: -113.99491753,
      time: 1578762000
    },
    name: "Baked And Loaded"
  }
}

export const mockRemovePotentialLocationActionObj = {
  type: 'REMOVE_POTENTIAL_LOCATION',
  potentialLocation: {
    desc: "Our food trailer is designed for efficiently producing quality comfort food! We serve baked potatoes loaded with a variety of toppings! BBQ Pulled Pork Tex Mex Mushroom Swiss Classic English Sheppard's Pie",
    image: undefined,
    location: {
      display: "Prairie Horizon Pop Up Market",
      latitude: 51.19624517,
      longitude: -113.99491753,
      time: 1578762000
    },
    name: "Baked And Loaded"
  }
}

export const mockChooseCity = {
  type: 'CHOOSE_CITY',
  city: 'test'
}

export const mockTruckAction = {
  type: 'TOGGLE_POTENTIAL_LOCATION',
  truck: {
    desc: "Our food trailer is designed for efficiently producing quality comfort food! We serve baked potatoes loaded with a variety of toppings! BBQ Pulled Pork Tex Mex Mushroom Swiss Classic English Sheppard's Pie",
    isPotentialLocation: true,
    image: undefined,
    location: {
      display: "Prairie Horizon Pop Up Market",
      latitude: 51.19624517,
      longitude: -113.99491753,
      time: 1578762000
    },
    name: "Baked And Loaded"
  }
}

export const mockCity = {
  city: 'calgary',
  lat: 51.0,
  long: -114.0
}

export const mockCityActionObj = {city: {city: "calgary", lat: 51, long: -114}, type: "CHOOSE_CITY"}
