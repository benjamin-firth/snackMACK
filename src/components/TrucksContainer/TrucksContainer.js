import React, { Component } from 'react';
import { connect } from 'react-redux';
import './TrucksContainer.scss';
import Header from '../Header/Header';
import Truck from '../Truck/Truck';
import PotentialLocations from '../PotentialLocations/PotentialLocations';
import TruckMap from '../Map/Map';
import { fetchFoodTrucks } from '../../utils/apiCalls';
import { setTrucks, togglePotentialLocation } from '../../actions';

export class TrucksContainer extends Component {
  constructor() {
    super();
    this.state = {
      showMap: true
    }
  }

  componentDidMount() {
    fetchFoodTrucks(this.props.city.city)
      .then(data => {
        const vendors = data.vendors;
        const vendorKeys = Object.keys(vendors);
        const nearbyTrucks = vendorKeys.map(key => {
          return { 
            name: vendors[key].name, 
            desc: vendors[key].description,
            location: vendors[key].last,
            image: vendors[key].images,
            email: vendors[key].url,
            isPotentialLocation: false
          }
        });
        this.props.setTrucks(nearbyTrucks);
      })
  }

  toggleMap = () => {
    this.setState({showMap: !this.state.showMap})
  }

  render() {
    const Trucks = this.props.allTrucks.map(truck => {
      return (
        <Truck 
          truck={truck}
          key={truck.name}
        />) 
    })

    return (
      <main>
        <Header />
        <div className='mainpage-container'>
          <div className='sidebar'>
            <button onClick={this.toggleMap}>{this.state.showMap? 'VIEW POTENTIAL LOCATIONS' : 'VIEW MAP OF TRUCKS'}</button>
            <div className='inner-scroll-container'>
              {Trucks}
            </div>
          </div>
          {this.state.showMap? (
            <TruckMap 
              lat={this.props.city.lat}
              long={this.props.city.long}
              trucks={this.props.allTrucks}
              togglePotentialLocation={this.props.togglePotentialLocation}
              key={this.props.city}
              />
          ) : <PotentialLocations />
          }
        </div>
      </main>
    )
  }
}


export const mapStateToProps = state => ({
  city: state.city,
  allTrucks: state.allTrucks
})

export const mapDispatchToProps = dispatch => ({
  setTrucks: trucks => dispatch(setTrucks(trucks)),
  togglePotentialLocation: truck => dispatch(togglePotentialLocation(truck))
})

export default connect(mapStateToProps, mapDispatchToProps)(TrucksContainer);