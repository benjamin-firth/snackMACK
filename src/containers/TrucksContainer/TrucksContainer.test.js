import React from 'react';
import { shallow } from 'enzyme';
import { TrucksContainer, mapStateToProps, mapDispatchToProps } from './TrucksContainer';
import { fetchFoodTrucks } from '../../utils/apiCalls';
import * as mockData from '../../utils/mockData';
import { setTrucks, togglePotentialLocation } from '../../actions';

jest.mock('../../utils/apiCalls');

describe('TrucksContainer', () => {
  let wrapper;

  beforeEach(() => {
    fetchFoodTrucks.mockImplementation(() => Promise.resolve(mockData.mockPotentialLocations));
    wrapper = shallow( <TrucksContainer city={{}} setTrucks={jest.fn()} togglePotentialLocation={jest.fn()} allTrucks={[]} city='test'/>)
  })

it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  describe('mapStateToProps', () => {
    it('should return an object with city and allTrucks info', () => {
      const mockState = {
        city: 'calgary',
        allTrucks: mockData.mockPotentialLocations
      };
      const expected = {
        city: 'calgary',
        allTrucks: mockData.mockPotentialLocations
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    })
  })

  describe('fetchFoodTrucks', () => {
    it('should call fetchFoodTrucks when componentDidMount is called', () => { 

      expect(fetchFoodTrucks).toHaveBeenCalled();
    })
  })

  describe('toggleMap', () => {
    it('should set state of showMap when toggleMap is called', async () => {
      wrapper.setState({
        showMap: false
      });
  
      await wrapper.instance().toggleMap();
  
      expect(wrapper.state('showMap')).toEqual(true);
    })
  })

  describe('mapDispatchToProps', () => {
    const mockDispatch = jest.fn();

    it('should should call setTrucks on componentDidMount', () => {
      const actionToDispatch = setTrucks(mockData.mockPotentialLocations);
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.setTrucks(mockData.mockPotentialLocations);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })

    it('should should call togglePotentialLocation on componentDidMount', () => {
      const actionToDispatch = togglePotentialLocation(mockData.mockToggleTruck);
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.togglePotentialLocation(mockData.mockToggleTruck);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })
  })

})