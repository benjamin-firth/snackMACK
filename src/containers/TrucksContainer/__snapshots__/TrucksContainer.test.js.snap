// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrucksContainer should match the snapshot 1`] = `
<main>
  <Header />
  <div
    className="mainpage-container"
  >
    <div
      className="sidebar"
    >
      <button
        onClick={[Function]}
      >
        VIEW POTENTIAL LOCATIONS
      </button>
      <div
        className="inner-scroll-container"
      />
    </div>
    <TruckMap
      key="test"
      trucks={Array []}
    />
  </div>
</main>
`;
